apm-agent-core/src/test/java/co/elastic/apm/impl/payload/TransactionPayloadJsonSchemaTest.java

```diff
@@ -113,6 +114,55 @@ void testJsonSchemaDslJsonAllValues() throws IOException {
         validate(createPayloadWithAllValues());
     }
 
+    @Test
+    void testJsonStructure() throws IOException {
+        TransactionPayload payload = createPayloadWithAllValues();
+
+        when(coreConfiguration.isDistributedTracingEnabled()).thenReturn(false);
+        DslJsonSerializer serializer = new DslJsonSerializer(coreConfiguration.isDistributedTracingEnabled(), mock(StacktraceConfiguration.class));
+
+        validateDbSpanSchema(payload, serializer, true);
+
+        for(Span span: payload.getTransactions().get(0).getSpans()) {
+            if(span.getType().equals("db.postgresql.query")) {
+                span.getContext().getTags().clear();
+                validateDbSpanSchema(payload, serializer, false);
+                break;
+            }
+        }
+    }
+
+    private void validateDbSpanSchema(TransactionPayload payload, DslJsonSerializer serializer, boolean shouldContainTags) throws IOException {
+        final String content = serializer.toJsonString(payload);
+        JsonNode node = objectMapper.readTree(content);
+
+        boolean contextOfDbSpanFound = false;
+        for (JsonNode child: node.get("transactions").get(0).get("spans")) {
```

Comment:
In intake v2, the spans are not nested inside transactions, but in `payload.spans`

Reply:
So what do you suggest?

Reply:
Look in both locations